### PR TITLE
[network] Make NetworkContext serialize consistently

### PR DESCRIPTION
**Pull request by gregnazario**
### Overview
There was a mismatch on serialization which caused issues in the underlying ES storage.

* Private was serializing like `{"network_id":{"private":"vfn"}}`
* Public was serializing like `{"network_id":"public"}`

Now all serialize to a string.  This does cause a problem if you name the private network `public` or `validator`.

### CODE REVIEW

File: config/src/network_id.rs [hunk 1]
```diff
@@ -2,19 +2,43 @@
...
 /// for logging accordingly.
-#[derive(Clone, Eq, PartialEq, Serialize)]
+#[derive(Clone, Eq, PartialEq)]
 pub struct NetworkContext {
     network_id: NetworkId,
     role: RoleType,
     peer_id: PeerId,
 }
 
+/// This custom serializer is to make sure that there is a consistent output type of
```
Comment (davidiw) [inline config/src/network_id.rs:18]: might be easier to use a serde field attribute serialize_with and just point to how you want the network_id to look. https://serde.rs/field-attrs.html

**Review by davidiw — APPROVED** [review]

File: config/src/network_id.rs (hunk 1 above)
Comment (gregnazario) [reply config/src/network_id.rs:18]: Yeah, that probably was easier :P

**Pull request closed without merging**